eng/dockerfile-templates/monitor/Dockerfile.linux

```
@@ -0,0 +1,55 @@
+{{
+    _ .NET major version matches the major version of dotnet-monitor ^
+    set dotnetMajor to split(PRODUCT_VERSION, ".")[0] ^
+    set dotnetMajorMinor to cat(dotnetMajor, ".0") ^
+    set monitorMajor to split(PRODUCT_VERSION, ".")[0] ^
+    set isAlpine to find(OS_VERSION, "alpine") >= 0 ^
+    set isMariner to find(OS_VERSION, "mariner") >= 0 ^
+    set isDistroless to find(OS_VERSION, "distroless") >= 0 || find(OS_VERSION, "chiseled") >= 0 ^
+    set isSingleStage to isAlpine ^
+    set aspnetBaseTag to
+        cat("$REPO:", VARIABLES[cat("dotnet|", dotnetMajorMinor, "|product-version")], "-", OS_VERSION, ARCH_TAG_SUFFIX) ^
+    set osVersionBase to when(isDistroless, match(OS_VERSION, ".+(?=.*-)")[0], OS_VERSION_BASE) ^
+    set installerImageTag to when(isMariner,
+        cat("mcr.microsoft.com/cbl-mariner/base/core:", OS_VERSION_NUMBER),
+        when(isAlpine,
+            aspnetBaseTag,
+            cat(ARCH_VERSIONED, "/buildpack-deps:", osVersionBase, "-curl")))
```

There should be no reason you can't use the aspnet base image as the base for everything. So this can be simplified to:

```suggestion
    set installerImageTag to aspnetBaseTag
```

Or just replace all usages of `installerImageTag` with `aspnetBaseTag`.

---

> There should be no reason you can't use the aspnet base image as the base for everything. 

What does using the aspnet base image provide that these base images don't? Is it mostly a convenience because the dockerfile already depends on it?

---

No functional difference. It just simplifies things, specifically the template.

---

Kindof... I think I would need to use the non-distroless ASP.NET image for the installer image when it's doing a multi-stage build of a distroless image.

---

I ended up removing '-distroless' and '-chiseled' from the OS_VERSION to determine the non-distroless OS and version and use that for the installer image. Let me know what you think.

---

Oh, totally forgot that we need to account for distroless.

---

Sigh, `curl` doesn't exist on the Ubuntu Jammy ASP.NET images. I think I'm going to revert to my original solution (the same thing that runtime and aspnet are doing with the buildpack-deps installer image).